modules/oe_whitelabel_extra_project/oe_whitelabel_extra_project.module

```diff
@@ -144,5 +160,92 @@ function _oe_whitelabel_extra_project_preprocess_inpage_nav(array &$variables):
       'label' => $node->{$item['#field_name']}->getFieldDefinition()->getLabel(),
     ];
   }
+}
 
+/**
+ * Adds variables for the project status.
+ *
+ * @param array $variables
+ *   Variables from hook_preprocess_node().
+ */
+function _oe_whitelabel_extra_project_preprocess_status_and_progress(array &$variables): void {
+  /** @var \Drupal\node\NodeInterface $node */
+  $node = $variables['node'];
+  /** @var \Drupal\datetime_range\Plugin\Field\FieldType\DateRangeItem|null $date_range_item */
+  $date_range_item = $node->get('oe_project_dates')->first();
+  if ($date_range_item === NULL) {
+    return;
+  }
+  $status_keys = ['planned', 'ongoing', 'closed'];
+  $status_labels = [t('Planned'), t('Ongoing'), t('Closed')];
+
+  if (!$date_range_item->value || !$date_range_item->end_value) {
+    // One of the fields is empty.
+    return;
+  }
+
+  // Dates only store the date, not the time.
+  // Use the site-wide configured timezone, not a user-specific timezone.
+  /* @see \Drupal\system\TimeZoneResolver::getTimeZone() */
+  /** @var string $timezone */
+  $timezone = \Drupal::config('system.date')->get('timezone.default') ?? 'UTC';
+  $get_timestamp = static function (string $date_string) use ($timezone): int {
+    return (new DrupalDateTime($date_string, $timezone))->getTimestamp();
+  };
+  // Project starts at the beginning of the first day at 00:00.
+  $t_start = $get_timestamp($date_range_item->value);
+  // Project ends at the end of the last day at 24:00.
+  $t_end = $get_timestamp($date_range_item->end_value . ' +1 day');
```

Comment:
A day is an interval. Perhaps we could call it an half-closed interval. So `[start...end[` or `[start...end)` in math notation, or `t >= t_start && t < t_end`. The `00:00:00 moment already belongs to the next day, but every moment nanoseconds before that is from the current day.
So the most semantically correct timestamp we can send is exactly the end of the day. It is not the last moment in the interval, but the upper limit. `sup`, not `max`.
Then it is up to the js what to do with it, to interpret it as an upper limit, not as a max. (but it doesn't make any different in practice I think)

Reply:
> it passes to the next day in the calendar, so if end date is 31st it passes to 1st of next month

this would be ok or not?
the timestamp is only used to calculate the progress bar and status label.

Reply:
btw 23:59:59 is incorrect for summer / winter time switch :)

Reply:
not sure about the summer saving but agreed that in this specific use case we can get away with the +1